src/main/java/proto/objects.proto

```diff
@@ -0,0 +1,30 @@
+/*
+Instead of encoding all data in one proto, we will instead preface the proto
+with a header field containing the number of total bytes in the proto. This will
+be a configurable value that will be computed where the biggest possible Data
+division below the maximum threshold is used. This will allow for easy chunking
+when reading in messages etc.
+*/
+syntax = "proto3";
+
+package formats;
+option java_package = "proto";
+option java_outer_classname = "DataFormats";
+
+message SerializedData { repeated Data data = 1; }
+message Data {
+  Position pose = 1;
```

Comment:
The pose is 6D, it containts position and orientation.
message Pose {
double lat = 1;
double lng = 2;
double alt = 3;
float roll_deg = 4;
float pitch_deg = 5;
float yaw_deg = 6;
}
And describe each field, what your xyz axis is.

Reply:
Could you elaborate what you mean describe each field, I think I'm not fluent enough in the terminology here to know what you have in mind?

Reply:
Add comment for the field meaning in your proto definition, e.g.:
message Pose {
// Position is in WGS84 latitude, longitude, altitude.
double lat = 1;  // Degrees in [-90, 90].
double lng = 2; // Degrees in [-180, 180).
double alt = 3;  // Meters over WGS84 datum.
// The orientation, what's the local pose frame? what's the x,y,z direction? what's the angular unit?
float roll_deg = 4;
float pitch_deg = 5;
float yaw_deg = 6;
}

Reply:
I think you need "optional", and you can also give the field a default value.